Load thruster parameters

Thomio-Watanabe opened the pull request:
This branch adds functionality to export the thruster explicitly. Now the user has to define the thruster he wants to export inside the <plugin> tag from the Gazebo model that contains the thrusters. The thruster name must be the same as the link name you want to use as thruster. This branch also creates the parameters: min_thrust and max_thrust to limit thrust values.

Thomio-Watanabe committed:
Load plugin parameters defined inside the tag <plugin>

Thomio-Watanabe committed:
bug-fix: use right sdfelement to read linked list

Thomio-Watanabe committed:
Add method checkThrustLimits

doudou commented on src/GazeboThruster.cpp:
```diff
@@ -26,60 +28,146 @@ void GazeboThruster::Load(physics::ModelPtr _model,sdf::ElementPtr _sdf)
-                + model->GetName();
-        gzthrow(msg);
+        bool thrusterFound = false;
+        for(physics::Link_V::iterator link = links.begin(); link != links.end(); ++link)
```
Can't you get a link by name ????

doudou commented on src/GazeboThruster.cpp:
```diff
@@ -15,9 +16,10 @@ GazeboThruster::~GazeboThruster()
+    gzmsg << "GazeboThruster: loading thrusters from model: " << model->GetName() << endl;
 
-    loadLinks();
+    loadThrusters();
```
This is a typical situation where side-effects are harmful.

In loadThrusters, you fill the thrusters vector

In checkThrusters, you verify that the thrusters vector is valid.

This means that if you have an invalid thruster definition, you will end up with a class with an invalid state.

The better option would be:

```
std::vector<> thrusters = loadThrusters();
if (checkThrusters(thrusters))
   this->thrusters = thrusters;
```

Thomio-Watanabe committed:
Refactor code:
 - make loadThrusters() and checkThrusters() methods side effect free
 - find link by name inside the method checkThrusters()

Thomio-Watanabe commented:
Refactor code
